File: 1-Two-Sum.java
/*
    Leetcode #1 - https://leetcode.com/problems/two-sum/
*/

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> hash = new HashMap<Integer, Integer>();

        for(int i = 0; i < nums.length; i++) {
            int key = target - nums[i];
            if(hash.containsKey(key)) {
                int[] twoSums = {hash.get(key), i};
                return twoSums;
            }
            else {
                hash.put(target - key, i);
            }
            
        }
        
        return null;
    }
}

File: Leetcode/1-two-sum.py
# Leetcode #1 - https://leetcode.com/problems/two-sum/

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash = {}
        for i, num in enumerate(nums):
            if num in hash:
                return [i, hash[num]]
            else:
                hash[target - num] = i